src/cmd/run.js

```diff
@@ -44,16 +47,18 @@ export type WatcherCreatorParams = {|
   desktopNotifications?: typeof defaultDesktopNotifications,
   ignoreFiles?: Array<string>,
   createFileFilter?: FileFilterCreatorFn,
+  addonReload?: typeof defaultAddonReload,
 |};
 
+
 export type WatcherCreatorFn = (params: WatcherCreatorParams) => Watchpack;
 
 export function defaultWatcherCreator(
   {
     addonId, client, sourceDir, artifactsDir, ignoreFiles,
```

Comment:
@saintsebastian actually I think that we can leave it like it is right now 

(initially I was thinking that we could pass an `addonReload` utility method that do not need to receive the `addonId` and the `client` from the `defaultWatcherCreator`, e.g. by capture them into a closure, which is absolutely possible, but after re-reading it again I changed my mind and I think that we can leave to `defaultWatcherCreator` the responsability of providing `addonId` and `client` to the `addonReload` function)